diff --git a/node_modules/react-native-paper/lib/module/components/BottomNavigation.js b/node_modules/react-native-paper/lib/module/components/BottomNavigation.js
index 22c4155..c64e179 100644
--- a/node_modules/react-native-paper/lib/module/components/BottomNavigation.js
+++ b/node_modules/react-native-paper/lib/module/components/BottomNavigation.js
@@ -344,13 +344,23 @@ class BottomNavigation extends React.Component {
     this.animateToCurrentIndex();
   }
 
+  componentDidMount() {
+    if (Platform.OS === 'ios') {
-      Keyboard.addListener('keyboardWillShow', this._handleKeyboardShow);
-      Keyboard.addListener('keyboardWillHide', this._handleKeyboardHide);
+      this.keyboardWillShow = Keyboard.addListener('keyboardWillShow', this._handleKeyboardShow);
+      this.keyboardWillHide = Keyboard.addListener('keyboardWillHide', this._handleKeyboardHide);
+    } else {
+      this.keyboardDidShow = Keyboard.addListener('keyboardDidShow', this._handleKeyboardShow);
+      this.keyboardDidHide = Keyboard.addListener('keyboardDidHide', this._handleKeyboardHide);
+    }
+  }
+
   componentWillUnmount() {
     if (Platform.OS === 'ios') {
+      this.keyboardWillShow?.remove();
+      this.keyboardWillHide?.remove();
     } else {
+      this.keyboardDidShow?.remove();
+      this.keyboardDidHide?.remove();
     }
   }
 
